src/EGraphs/egraph.jl

```diff
@@ -142,11 +141,10 @@ function EGraph{ExpressionType,Analysis}(; needslock::Bool = false) where {Expre
   EGraph{ExpressionType,Analysis}(
     UnionFind(),
     Dict{IdKey,EClass{Analysis}}(),
-    Vector{VecExpr}(),
     Dict{VecExpr,Id}(),
     Dict{UInt64,Any}(),
-    Id[],
-    UniqueQueue{Id}(),
+    Vector{Id}(),
+    Vector{Id}(),
```

Comment:
Why no more `UniqueQueue`?

Reply:
Not needed anymore. 

Reasoning: 
- We need the nodes _and_ eclass ids in the pending / analysis_pending vectors for rebuilding.
- Nodes may be canonicalized (while they are contained in the pending collections)
- UniqueQueue detects duplicates only when adding new entries, and it would not remove duplicates of nodes after canonicalization. 
- the rewrite of the rebuilding code removes duplicates in the pending and analysis_pending vectors after each iteration instead.